flake.nix

```diff
@@ -0,0 +1,59 @@
+{
+  description = "Solana development setup with Nix.";
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/c5e2528c7c4ec05ce05a563e3be64f3525b278ad";
+    flake-parts.url = "github:hercules-ci/flake-parts/f4330d22f1c5d2ba72d3d22df5597d123fdb60a9";
+    rust-overlay.url = "github:oxalica/rust-overlay/87f0965f9f5b13fca9f38074eee8369dc767550d";
+    crane.url = "github:ipetkov/crane";
+  };
+  outputs = inputs @ {
+    self,
+    nixpkgs,
+    flake-parts,
+    rust-overlay,
+    crane,
+  }:
+    flake-parts.lib.mkFlake {inherit inputs;} {
+      systems = [
+        "x86_64-linux"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "aarch64-darwin"
+        "x86_64-windows"
```

Comment:
do we know if this'll actually work on all these systems?

Reply:
Added CI workflow!